### DISCUSSION TITLE
Compilation error after adding KSPlayer to project that is already using github.com/arthenica/ffmpeg-kit

### DISCUSSION THREAD

**Discussion by mazz** [kingslay/KSPlayer#790] [General]
I need arthenica/ffmpeg-kit to call `ffmpeg` and `ffprobe` through their API. Is there a workaround? Perhaps I can use your FFMpegKit to call ffmpeg and ffprobe?

**Comment by kingslay** [comment]
FFMpegKit  在macOS上可以运行 swift run ffprobe  和 swift run ffmpeg

**Reply by mazz** [reply]
Can you show me an example how you use `swift run ffprobe` and `swift run ffmpeg`? It might be an adequate substitution but I will be losing the convenient async callbacks that arthenica/ffmpeg-kit currently gives me ...

**Comment by kingslay** [comment]
这个是要在终端命令行来执行命令，无法在代码上执行。

**Reply by mazz** [reply]
Would it be possible for you to rename your ffmpegkit library? I cannot get around the conflict that it creates with https://github.com/arthenica/ffmpeg-kit.git

arthenica/ffmpeg-kit.git is a very popular project with over 4100 stars.

**Comment by kingslay** [comment]
你是想要用ffmpeg-kit的那个功能呢？你可以把对应的代码发出来，我增加下。你想要的这个功能是指在macos上使用吗？还是ios和tvos也是需要使用这个功能呢？

**Reply by mazz** [reply]
here is an example of what I need to do. I need to do things like transcode an mp4, so I need to parse an ffmpeg command and execute it asynchronously like this:

```swift
                let session: FFmpegSession = FFmpegKit.executeAsync("-i \"\(symlinkUrl.path)\" -s hd720 -c:v libx264 -crf 27 -c:a aac -strict -2 \"\(transcodedPath)\"") { session in
                    // ...
                }
```

This is not trivial.

**Comment by kingslay** [comment]
我在lgpl版本增加了ffprobe_execute 和ffmpeg_execute。这样你就可以直接用代码调用ffmpeg和ffprobe了

**Reply by mazz** [reply]
> 我在lgpl版本增加了ffprobe_execute 和ffmpeg_execute。这样你就可以直接用代码调用ffmpeg和ffprobe了

Interesting, so `ffmpeg_execute` can replicate this functionality here?

```swift 
                let session: FFmpegSession = FFmpegKit.executeAsync("-i \"\(symlinkUrl.path)\" -s hd720 -c:v libx264 -crf 27 -c:a aac -strict -2 \"\(transcodedPath)\"") { session in
                    // ...
                }
```

Another issue is that the project I'm working on is non-profit. I cannot afford to pay for the LGPL library. This leaves me with a difficult choice because I either have to support FFMpegKit and find a sub-par video player OR use KSPlayer and sacrifice the ability to transcode. 

Wouldn't it be more equitable for you to simply stop having your ffmpeg sub-project conflict with arthenica/ffmpeg-kit, which is a pre-existing, very largely-used and supported project?
